actors/src/test/kotlin/kts/actors/actionsTest.kt

```diff
@@ -85,44 +85,47 @@ class ActionsTest {
   }
 
   @Test
-  fun `should not mutate multiple actions with then`() {
+  fun `action then sequence should chain underling actions into sequence`() {
     val firstAction = MockAction()
     val secondAction = MockAction()
     val thirdAction = MockAction()
 
-    val firstSequence = firstAction then secondAction
-    firstSequence then thirdAction // <-- should not mutate firstSequence
+    val sequence = firstAction then Actions.sequence(secondAction, thirdAction)
 
-    assertEquals(2, firstSequence.actions.size)
-    assertEquals(firstAction, firstSequence.actions[0])
-    assertEquals(secondAction, firstSequence.actions[1])
+    assertEquals(firstAction, sequence.actions[0])
+    assertEquals(secondAction, sequence.actions[1])
+    assertEquals(thirdAction, sequence.actions[2])
+    assertEquals(3, sequence.actions.size)
   }
 
   @Test
-  fun `should add actions into sequences`() {
+  fun `sequence then sequence should chain underling actions into sequence`() {
     val firstAction = MockAction()
     val secondAction = MockAction()
+    val thirdAction = MockAction()
+    val fourthAction = MockAction()
 
-    val sequence = firstAction + secondAction
+    val sequence = Actions.sequence(firstAction, secondAction) then Actions.sequence(thirdAction, fourthAction)
 
     assertEquals(firstAction, sequence.actions[0])
     assertEquals(secondAction, sequence.actions[1])
-    assertEquals(2, sequence.actions.size)
+    assertEquals(thirdAction, sequence.actions[2])
+    assertEquals(fourthAction, sequence.actions[3])
+    assertEquals(4, sequence.actions.size)
   }
 
   @Test
-  fun `should add multiple actions into sequences`() {
+  fun `should not mutate multiple actions with then`() {
```

Comment:
Not sure if this test is necessary. You could add extra asserts to existing tests with sequences.

Reply:
This would mix the responsibilty of the tests IMHO. One test make sure one can create a sequence with `then`. And one test make sure that `then` do not mutate (anymore).

But it is personal perference. Please confirm if you want me to change it.

Reply:
No, this is OK.